service/header/sync.go

```diff
@@ -70,10 +97,10 @@ func (s *Syncer) Validate(ctx context.Context, p peer.ID, msg *pubsub.Message) p
 		return pubsub.ValidationReject
 	}
 
-	// if syncing is still in progress - just ignore the new Header
-	// Syncer will fetch it after anyway
-	select {
-	case <-s.done:
+	// if syncing is still in progress - just ignore the new header as
+	// Syncer will fetch it after anyway, but if syncer is done, append
+	// the header.
```

Comment:
This is not trivial to understand but I might be just tired lol. The header will not be appended to the store because the syncer will fetch it again anyways sounds inefficient and confusing at first but I guess it makes sense.

Reply:
Yes it is. There is a todo for that and we will change it. A lot of sacrifices were made to make devnet run.

Reply:
Side note: @Wondertan and I discussed - validation function should be separated out of syncer because that's not the only component using validation logic.

Reply:
I think that makes sense.